### PR TITLE
London support

### PR DESCRIPTION
**Pull request by yann300**
ask the user to input maxFee and maxPriorityFee if the chain is eip 1559 enabled.
references for the implementation:
https://hackmd.io/4YVYKxxvRZGDto7aq7rVkg?view
https://github.com/ethereum/eth1.0-specs/blob/master/network-upgrades/ecosystem-readiness.md

EDIT: the last set of changes updates remix-ide to it default to London when the app starts
EDIT: this commit 533d7f9caf61739ea9ec4c1d3c20e86c880f2c63 `use berlin` roll back unit testing to use berlin, fix for making sure tests run with london will come later

### CODE REVIEW

File: apps/remix-ide/src/app/ui/confirmDialog.js [hunk 1]
```diff
@@ -74,10 +108,14 @@ function confirmDialog (tx, amount, gasEstimation, self, newGasPriceCb, initialP
     if (txFeeText) {
       el.querySelector('#txfee').innerHTML = txFeeText
     }
-    if (gasPriceValue) {
+    if (el.querySelector('#gasprice') && gasPriceValue) {
       el.querySelector('#gasprice').value = gasPriceValue
       onGasPriceChange()
     }
+    if (el.querySelector('#maxfee') && network && network.lastBlock && network.lastBlock.baseFeePerGas && el.querySelector('#maxfee')) {
```
Comment (yann300) [inline apps/remix-ide/src/app/ui/confirmDialog.js:115]: code duplication

File: apps/remix-ide/src/blockchain/execution-context.js [hunk 2]
```diff
@@ -176,6 +177,7 @@ export class ExecutionContext {
         const block = await web3.eth.getBlock('latest')
         // we can't use the blockGasLimit cause the next blocks could have a lower limit : https://github.com/ethereum/remix/issues/506
         this.blockGasLimit = (block && block.gasLimit) ? Math.floor(block.gasLimit - (5 * block.gasLimit) / 1024) : this.blockGasLimitDefault
+        this.lastBlock = block
```
Comment (yann300) [inline apps/remix-ide/src/blockchain/execution-context.js:180]: `_updateChainContext` should be updated as soon as possible when a new provider is switched to.

File: libs/remix-astwalker/package.json [hunk 3]
```diff
@@ -34,14 +34,14 @@
     ]
   },
   "dependencies": {
-    "@ethereumjs/block": "^3.3.0",
-    "@ethereumjs/tx": "^3.2.1",
-    "@ethereumjs/vm": "^5.4.1",
+    "@ethereumjs/block": "^3.4.0",
+    "@ethereumjs/tx": "^3.3.0",
+    "@ethereumjs/vm": "^5.5.0",
```
Comment (yann300) [inline libs/remix-astwalker/package.json:39]: might need to check what to do with `maxFee` and `maxPriorityFee`.

File: apps/remix-ide/src/blockchain/execution-context.js (hunk 2 above)
Comment (yann300) [reply apps/remix-ide/src/blockchain/execution-context.js:180]: the https://github.com/ethereum/remix-project/pull/1412/files/4944a397f4e8222aa6c9f3ac6e1ac1247a851c79#diff-e25e2a645b7e5c5e40e6acd6a0f942ddca48aac37521c89b1e514c3f3d582c7fL146 is already there

**Review by Aniket-Engg — COMMENTED** [review]
I think `JavaScript VM (London)` should be first option in `Environment` dropdown in Deploy and Run

File: apps/remix-ide/src/app/ui/confirmDialog.js [hunk 4]
```diff
@@ -14,20 +14,36 @@ var css = csjs`
...
+      el.gasPriceStatus = priceStatus
+      el.txFee = { maxFee, maxPriorityFee, baseFeePerGas: network.lastBlock.baseFeePerGas }
+    })
+  }
+
+  const el = yo`
     <div>
-      <div>You are about to create a transaction on the Main Network. Confirm the details to send the info to your provider.
-        <br>The provider for many users is MetaMask. The provider will ask you to sign the transaction before it is sent to the Main Network.</div>
+      <div>You are about to create a transaction on ${network.name}. Confirm the details to send the info to your provider.
```
Comment (LianaHus) [inline apps/remix-ide/src/app/ui/confirmDialog.js:45]: maybe "...on ${network.name} network. ..." would be the correct  info

File: apps/remix-ide/src/app/ui/confirmDialog.js [hunk 5]
```diff
@@ -14,20 +14,36 @@ var css = csjs`
...
+        <br>The provider for many users is MetaMask. The provider will ask you to sign the transaction before it is sent to ${network.name}.</div>
```
Comment (LianaHus) [inline apps/remix-ide/src/app/ui/confirmDialog.js:46]: same here

**Comment by LianaHus** [comment]
Some notes about UI changes after a discussion/call (@Aniket-Engg @LianaHus)
- Code goes before all gas calculations.
- Max tx fee goes the bottom and if possible should be highlighted to catch the attention as the most important info.
- Error about minimum fee should be either orange or red or should be replaced with validation(should be decided later)
In any case, it will disable the 'Confirm' button(?)

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**